tests/diff_test.sh

```diff
@@ -39,6 +39,21 @@ function test_diff_side_by_side()
 
 }
 
+function test_diff_folders()
+{
+  local folder_1="$SAMPLES_DIR/db_files"
+  local folder_2="$SAMPLES_DIR/first_set_of_bytes_from_disk"
+  local diff_cmd="diff --color=always -rq $folder_1 $folder_2 | grep $folder_1"
+
+  output=$(diff_folders 'an_invalid_file' "$folder_2")
+  ret="$?"
+  assertEquals "($LINENO) Expected 22" "22" "$ret"
+
+  output=$(diff_folders 'an_invalid_file' 'another_invalid_file')
+  ret="$?"
+  assertEquals "($LINENO) Expected 22" "22" "$ret"
+}
```

Comment:
Could you add some tests for valid cases? You can create some fake folders and test the function by calling it and comparing the output.